pkg/test/framework/components/apps/api/apps.go

```diff
@@ -0,0 +1,25 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package api
+
+import (
+	"istio.io/istio/pkg/test/framework/environment"
+)
+
+// Apps defines an interface for a bundle of test services.
+type Apps interface {
+	// GetApp returns a fake testing app object for the given name.
+	GetApp(name string) (environment.DeployedApp, error)
```

Comment:
Is this an open set? Would make sense to have methods for each app with a given name?

Reply:
meh, not too worried about it ATM. This makes the code in the environment simpler.